test/commands/user_track/generate_concepts_status_test.rb

```diff
@@ -0,0 +1,201 @@
+require "test_helper"
+
+class UserTrack::GenerateConceptsStatusTest < ActiveSupport::TestCase
+  test "generates concepts for empty track" do
+    track = create :track
+    user_track = create :user_track, track: track
+
+    assert_equal(
+      {},
+      ::UserTrack::GenerateConceptsStatus.(user_track)
+    )
+  end
+
+  test "concepts without prerequisite are :unlocked" do
+    track = create :track
+    user_track = create :user_track, track: track
+
+    create :track_concept, slug: 'basics', track: track
+
+    assert_equal(
+      { 'basics' => :unlocked },
+      ::UserTrack::GenerateConceptsStatus.(user_track)
+    )
+  end
+
+  test "concepts following incomplete concepts are :locked" do
+    track = create :track
+    user_track = create :user_track, track: track
+
+    basics = create :track_concept, slug: 'basics', track: track
+    booleans = create :track_concept, slug: 'booleans', track: track
+    atoms = create :track_concept, slug: 'atoms', track: track
+
+    lasagna = create :concept_exercise, track: track
+    lasagna.taught_concepts << basics
+
+    pacman = create :concept_exercise, track: track
+    pacman.taught_concepts << booleans
+    pacman.prerequisites << basics
+
+    logger = create :concept_exercise, track: track
+    logger.taught_concepts << atoms
+    logger.prerequisites << booleans
+
+    assert_equal(
+      {
+        'basics' => :unlocked,
+        'booleans' => :locked,
+        'atoms' => :locked
+      },
+      ::UserTrack::GenerateConceptsStatus.(user_track)
+    )
+  end
+
+  test "concepts following complete concepts are :unlocked" do
+    track = create :track
+    user_track = create :user_track, track: track
+
+    # Set up concepts
+
+    basics = create :track_concept, slug: 'basics', track: track
+    booleans = create :track_concept, slug: 'booleans', track: track
+    atoms = create :track_concept, slug: 'atoms', track: track
+
+    # Set up exercises
+
+    lasagna = create :concept_exercise, track: track
+    lasagna.taught_concepts << basics
+
+    pacman = create :concept_exercise, track: track
+    pacman.taught_concepts << booleans
+    pacman.prerequisites << basics
+
+    logger = create :concept_exercise, track: track
+    logger.taught_concepts << atoms
+    logger.prerequisites << booleans
+
+    # learn concepts
+
+    create :user_track_learnt_concept, user_track: user_track, concept: basics
+
+    assert_equal(
+      {
+        'basics' => :complete,
+        'booleans' => :unlocked,
+        'atoms' => :locked
+      },
+      ::UserTrack::GenerateConceptsStatus.(user_track)
```

Comment:
You shouldn't need the `::` at the start of these.

Reply:
Ok, I saw it elsewhere in another test file so though that was the codebase style, can be removed

Reply:
The only place it should be needed is if there is a potential conflict between two namespaces. I'm not sure why it's in `app/commands/user_track/create.rb` - feel free to remove it from there too :)

The example in `config/environments/production.rb` is a good one. Rails defines it's own Logger class, so here we're saying that we want to use the Ruby one, not the Rails one (ie, not the one in our current namespace).